native/dll/DX93D/SageTVDX93D.cpp

```diff
@@ -300,22 +300,25 @@ JNIEXPORT void JNICALL Java_sage_DirectX9SageRenderer_asyncVideoRender0
 		return;
 	}
 	const char* cName = env->GetStringUTFChars(sharedMemPrefix, NULL);
-	HANDLE fileMap = CreateFileMapping(INVALID_HANDLE_VALUE, NULL, PAGE_READWRITE,
-		0, 1920*540*3 + 1024, cName);
+	HANDLE fileMap = CreateFileMapping(INVALID_HANDLE_VALUE, NULL, PAGE_READWRITE, 0, 1920 * 540 * 3 + 1024, cName);
+	if (fileMap == NULL)
+		return;
 	char buf[256];
 	asyncRenderCount++;
 	Sleep(100); // wait more than 50 msec so if a prior MPlayer process was killed forcibly the loop below would have
 	// had time to exit and cleanup its pointers before we start on our work
-	strcpy(buf, cName);
-	strcat(buf, "FrameReady");
+	strcpy_s(buf, sizeof(buf), cName);
+	strcat_s(buf, sizeof(buf), "FrameReady");
 	HANDLE evtReady = CreateEvent(NULL, FALSE, FALSE, buf);
-	strcpy(buf, cName);
-	strcat(buf, "FrameDone");
+	strcpy_s(buf, sizeof(buf), cName);
+	strcat_s(buf, sizeof(buf), "FrameDone");
 	HANDLE evtDone = CreateEvent(NULL, FALSE, FALSE, buf);
 	env->ReleaseStringUTFChars(sharedMemPrefix, cName);
 	slog((env, "Created FileMap=0x%p evtReady=0x%p evtDone=0x%p\r\n", fileMap, evtReady, evtDone));
 	unsigned char* myPtr = (unsigned char*)MapViewOfFile(fileMap, FILE_MAP_READ|FILE_MAP_WRITE, 0, 0, 0);
 	unsigned int* myData = (unsigned int*) myPtr;
+	if (myData == NULL)

```

Comment:
I'd change this to be 'if (myPtr == NULL || evtReady == NULL || evtDone == NULL)' and then put a block below like at the end of the file where you release all of the non-null objects so they get properly cleaned up in case of error. Then get rid of your other NULL checks below.

Reply:
OK.  Done